Integrations/integration-Palo_Alto_AppFramework.yml

```diff
@@ -0,0 +1,717 @@
+
+        args = demisto.args()
+
+        start_time = args.get('startTime')
+        end_time = args.get('endTime')
+        value = args.get('logsAmount')
+        time_range = args.get('timeRange')
+        time_value = args.get('rangeValue')
+
+        if (time_range):
+            if (time_value):
+                service_end_date = datetime.datetime.now()
+                service_start_date = get_start_time(time_range, int(time_value))
+            else:
+                raise 'Please enter timeRange and timeValue, or startTime and endTime'
+        else:
+            # parses user input to datetime object
+            service_start_date = datetime.datetime.strptime(start_time, "%Y-%m-%d %H:%M:%S")
+            service_end_date = datetime.datetime.strptime(end_time, "%Y-%m-%d %H:%M:%S")
+
+        # transforms datetime object to epoch time
+        service_start_date_epoch = int(service_start_date.strftime("%s"))
+        service_end_date_epoch = int(service_end_date.strftime("%s"))
+
+        api_query = "SELECT * FROM panw.threat WHERE severity = '5' LIMIT " + value
+
+        query_data = {
+            "query": api_query,
+            "startTime": service_start_date_epoch,
+            "endTime": service_end_date_epoch,
+        }
+        pages = query_loggings(query_data)
+
+        results = []
+        table_name = None
+        for page in pages:
+            current_page = page.json()
+            table_name = current_page['result']['esQuery']['table'][0].split('.')[1]
+            table_rows = current_page['result']['esResult']['hits']['hits']
+
+            for row in table_rows:
+                row_id = row['_id']
+                row_contents = row['_source']
+                row_contents['id'] = row_id
+                row_contents['score'] = row['_score']
+                results.append(row_contents)
+
+        screened_results = results_screener('threat', results)
+        # screened_results=results
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': results,
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown('Logs ' + table_name + ' table',  screened_results),
+            'EntryContext': {
+                  'Appframework.Logging(val.id==obj.id)': results
+            }
+        }
+        return entry
+
+    def get_social_applications_command():
+        ''' corresponds to 'pan-appframework-get-social-applications' command. Queries Appframework Logging according to a pre-set query '''
+
+        args = demisto.args()
+
+        start_time = args.get('startTime')
+        end_time = args.get('endTime')
+        value = args.get('logsAmount')
+        time_range = args.get('timeRange')
+        time_value = args.get('rangeValue')
+
+        if (time_range):
+            if (time_value):
```

Comment:
It would have the same effect. timeRange & timeValue are translated to datetime, and default returns the last X values in either case. The actual limit is the query's LIMIT.

Reply:
ok